<a href="https://colab.research.google.com/github/AdnanAli-10/Airbnb-Bookings-Analysis-Capstone-Project./blob/main/Airbnb_Bookings_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>

## <b>Defining a hypothetical situation</b>
Let us assume that my dad is a wealthy individual and is looking to make some investments in real estate. He has heard from his colleagues that the New York City real estate market has been thriving and makes good enough ROI(return on investment). Being in India, my dad is quite skeptical about investing in New York and leaving the property abandoned, He also wants some cash flow to be generated out of the property every now and then. To get out of this dilemma my dad calls his old friend, Mr Robert Kiyosaki (author of the book Rich Dad Poor Dad), he advices my dad to buy the property and list it on some rental services like <b>"Airbnb"</b>. My dad after listening to his friend comes and asks me <b>"Son, what is Airbnb?"</b>

<b>Defining Airbnb - Airbnb is an online marketplace connecting travelers with local hosts. On one side, the platform enables people to list their available space and earn extra income in the form of rent. On the other, Airbnb enables travelers to book unique homestays from local hosts, saving them money and giving them a chance to interact with locals. Catering to the on-demand travel industry, Airbnb is present in over 190 countries across the world.</b>

After this professional explanation to my dad, he is quite happy about the opportunity his friend has opened up for him. Now he wants answers to his questions from the business point of view. He asks to me find out a few unique details about the New York Airbnb market, for example:- Which part of New York generates the most revenue, etc. Now, me  being a data scientist I decide to tackle this problem through the art of Data Science and make my Indian parent proud(Probably the most hypothetical thing I've mentioned till now). 

# <b>So lets begin!!!</b>

## <b> Questions to answer:</b>
1. For which location do the customers pay the highest and lowest rent.
2. Top 5 highest listing areas/locations.
3. What is the average preferred price by the customer.
4. What are the total number of nights spent per location.
5. Which hosts are busiest and why.

Now that we have the questions, it is time to find the answers, but as the great Sherlock Holmes says - "It is a capital mistake to theorize before one has data.", So lets try finding some data.
In this hypothetical world, the organisation I'm working for (AlmaBetter) is kind enough to research and offer me the data required for the analysis. It can also be fetched from kaggle, named - Airbnb-NYC-cleaned by Sandeep Majumdar(The data might be a bit different so the results may vary). 

In [2]:
#Mounting the drive
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
#Importing Dependencies
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
plt.rcParams['figure.figsize'] = (10, 7)

In [4]:
#Reading the data in a dataframe using pandas library to perform exploratory analysis
airbnb_df = pd.read_csv('/content/drive/MyDrive/Airbnb NYC 2019.csv')

## <b>Initial EDA</b>

In [5]:
#checking the shape of the dataframe
airbnb_df.shape

(48895, 16)

In [14]:
airbnb_df.room_type.value_counts()

Entire home/apt    25409
Private room       22326
Shared room         1160
Name: room_type, dtype: int64

In [6]:
#Checking the first 5 values of the dataframe
airbnb_df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [7]:
#Checking the last 5 values of the dataframe
airbnb_df.tail()

id                                               name   host_id  \
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23

In [12]:
#Checking the columns
col_list = airbnb_df.columns.tolist()
col_list

['id',
 'name',
 'host_id',
 'host_name',
 'neighbourhood_group',
 'neighbourhood',
 'latitude',
 'longitude',
 'room_type',
 'price',
 'minimum_nights',
 'number_of_reviews',
 'last_review',
 'reviews_per_month',
 'calculated_host_listings_count',
 'availability_365']

<b> Defining the columns:</b>
We can see that we have 16 columns and 48,895 observations. To better understand the dataset let's see what each column means.
* id : A unique id given to each airbnb lisitng.
* name : The Ad title for the listing on Airbnb website.
* host_id : A unique id given to an Airbnb host.
* host_name : The name with which the host is registered.
* neighbourhood_group : A group of areas/neighbourhoods.
* neighbourhood : Name of a particular area/ neighbourhood.
* latitude : latitudinal coordinate of the listing.
* longitude : longitudinal coordinate of listing.
* room_type : listing type(1 of 3 types) - 1.Entire Home/apartment, 2.Private room, 3.Shared room.
* price : price of the listing.
* minimum_nights : Minimum number of nights required to stay in a single visit.
* number_of_reviews : The total number of reviews given by visitors.
* last_review : date of the last recorded review.
* reviews_per_month : The number of reviews given per month for a listing.
* calculated_host_listings_count : the total number of listings registered under a given host.
* availability_365 : the number of days for which a listing is available in a year.

In [16]:
#general info about the dataset
airbnb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

Now I will be removing a few columns (last_review, reviews_per_month, latitude and longitude), because I don't see them adding any value to the questions I have to answer. 

In [ ]:
#removing columns
airbnb_df.drop(['latitude','longitude','last_review','reviews_per_month'],axis=1,inplace=True)

In [26]:
#checking for null values
airbnb_df.isnull().sum()

id                                 0
name                              16
host_id                            0
host_name                         21
neighbourhood_group                0
neighbourhood                      0
room_type                          0
price                              0
minimum_nights                     0
number_of_reviews                  0
calculated_host_listings_count     0
availability_365                   0
dtype: int64

<b> Dealing with null values: </b>
There are two ways of dealing with null values, one is deleting the null vlaues and the other is to impute the null values with some meaningfull values.

In this case I will just delete the null values because they are not many and deleting them will not affect the trends much.

In [29]:
#Deleting the observations with null values
airbnb_df.dropna(inplace=True)

In [28]:
airbnb_df.isnull().sum()

id                                0
name                              0
host_id                           0
host_name                         0
neighbourhood_group               0
neighbourhood                     0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
calculated_host_listings_count    0
availability_365                  0
dtype: int64

We can see that there are no more null values left in the dataset.

In [32]:
#Getting a statistical summary of the dataframe
airbnb_df.describe()

id       host_id         price  minimum_nights  \
count  4.885800e+04  4.885800e+04  48858.000000    48858.000000   
mean   1.902335e+07  6.763169e+07    152.740309        7.012444   
std    1.098289e+07  7.862389e+07    240.232386       20.019757   
min    2.539000e+03  2.438000e+03      0.000000        1.000000   
25%    9.475980e+06  7.818669e+06     69.000000        1.000000   
50%    1.969114e+07  3.079133e+07    106.000000        3.000000   
75%    2.915765e+07  1.074344e+08    175.000000        5.000000   
max    3.648724e+07  2.743213e+08  10000.000000     1250.000000   

       number_of_reviews  calculated_host_listings_count  availability_365  
count       48858.000000                    48858.000000      48858.000000  
mean           23.273098                        7.148369        112.801425  
std            44.549898                       32.964600        131.610962  
min             0.000000                        1.000000          0.000000  
25%             1.000000                        1.000000          0.000000  
50%             5.000000                        1.000000         45.000000  
75%            24.000000                        2.000000        227.000000  
max           629.000000                      327.000000        365.000000

Here we can see that there are price values which are 0. This doesn't make sense because I don't think so people would put up listing for free. We will impute values for these wrong observations.

In [41]:
airbnb_df[airbnb_df['price']==0]

id                                               name    host_id  \
23161  18750597  Huge Brooklyn Brownstone Living, Close to it all.    8993084   
25433  20333471      ★Hostel Style Room | Ideal Traveling Buddies★  131697576   
25634  20523843    MARTIAL LOFT 3: REDEMPTION (upstairs, 2nd room)   15787004   
25753  20608117                    Sunny, Quiet Room in Greenpoint    1641537   
25778  20624541      Modern apartment in the heart of Williamsburg   10132166   
25794  20639628  Spacious comfortable master bedroom with nice ...   86327101   
25795  20639792  Contemporary bedroom in brownstone with nice view   86327101   
25796  20639914       Cozy yet spacious private brownstone bedroom   86327101   
26259  20933849                              the best you can find   13709292   
26841  21291569  Coliving in Brooklyn! Modern design / Shared room  101970559   
26866  21304320             Best Coliving space ever! Shared room.  101970559   

          host_name neighbourhood_group       neighbourhood        room_type  \
23161      Kimberly            Brooklyn  Bedford-Stuyvesant     Private room   
25433        Anisha               Bronx     East Morrisania     Private room   
25634  Martial Loft            Brooklyn            Bushwick     Private room   
25753        Lauren            Brooklyn          Greenpoint     Private room   
25778       Aymeric            Brooklyn        Williamsburg  Entire home/apt   
25794       Adeyemi            Brooklyn  Bedford-Stuyvesant     Private room   
25795       Adeyemi            Brooklyn  Bedford-Stuyvesant     Private room   
25796       Adeyemi            Brooklyn  Bedford-Stuyvesant     Private room   
26259        Qiuchi           Manhattan         Murray Hill  Entire home/apt   
26841        Sergii            Brooklyn            Bushwick      Shared room   
26866        Sergii            Brooklyn            Bushwick      Shared room   

       price  minimum_nights  number_of_reviews  \
23161      0               4                  1   
25433      0               2                 55   
25634      0               2                 16   
25753      0               2                 12   
25778      0               5                  3   
25794      0               1                 93   
25795      0               1                 95   
25796      0               1                 95   
26259      0               3                  0   
26841      0              30                  2   
26866      0              30                  5   

       calculated_host_listings_count  availability_365  
23161                               4                28  
25433                               4               127  
25634                               5                 0  
25753                               2                 0  
25778                               1                73  
25794                               6               176  
25795                               6               232  
25796                               6               222  
26259                               1                 0  
26841                               6               333  
26866                               6               139

In [37]:
airbnb_df[airbnb_df['price']==0].shape

(11, 12)

So there are 11 observations with price as 0 which need to be treated.

In [40]:
airbnb_df.groupby('minimum_nights')['price'].mean().reset_index()

minimum_nights       price
0                 1  142.062756
1                 2  146.279374
2                 3  160.285643
3                 4  161.229603
4                 5  157.263765
..              ...         ...
103             400   50.000000
104             480  199.000000
105             500   88.800000
106             999   96.000000
107            1250  180.000000

[108 rows x 2 columns]